### ISSUE TITLE
Scope changes to Porter for upcoming PREApplication/Payment Changes

### ISSUE THREAD

**Issue by derekpierre** [nucypher/nucypher#2843]
## `/get_ursulas`

The current `get_ursulas` Porter endpoint for Ursula sampling is defined as:

```
{
    "quantity": <int>,
    "duration_periods": <int>,
    "include_ursulas": List[String],   # optional
    "exclude_ursulas": List[String],   # optional
}
```

Presumably, without the concept of periods `duration_periods` would become `start_timestamp` and `end_timestamp` each represented as unix epoch ints. `start_timestamp` would be optional, so if not specified, the policy is to start immediately. So:
```
{
    "quantity": <int>,
    "start_timestamp": <int>,  # optional
    "end_timestamp": <int>,
    "include_ursulas": List[String],   # optional
    "exclude_ursulas": List[String],   # optional
}
```

## `/retrieve_cfrags`

The current `retrieve_cfrags` Porter endpoint for Bob making re-encryption requests is defined as:

```
{
        "treasure_map": <base64 string>,
        "retrieval_kits": List[<base64 string>],
        "alice_verifying_key": <hex string>,
        "bob_encrypting_key": <hex string>,
        "bob_verifying_key": <hex string>
}
```

My assumption here is that even though Ursula will need to know what network (Polygon initially, but others later on) to use to verify the policy is valid/paid for, that network information will be included in the treasure map. See comment [here](https://github.com/nucypher/nucypher/pull/2840#issuecomment-1011281269). If not, then this API will need to add something like a `policy_payment_network` parameter. The treasure map makes sense to me, but interested in others thoughts here.

## `/revoke`

There is currently no off-chain revoke functionality (only on-chain) for `nucypher` PRE, so Porter does not have a `/revoke` endpoint. Will this change?

**Comment by derekpierre** [comment]
^ @piotr-roslaniec , @KPrasch , @cygnusv

**Comment by derekpierre** [comment]
Initially addressed via #2857. Any required follow-up issues can be created from that base.
